mock/mock.go

```diff
@@ -0,0 +1,16 @@
+package mock
+
+import "github.com/stretchr/testify/mock"
+
+type Client struct {
+	mock.Mock
+}
+
+func NewClient() *Client {
```

Comment:
This `NewClient()` returns a pointer, whereas the "normal" one returns a non-pointer. Is that difference intentional?